samples/csharp/MicrosoftGraphSDK/Assignment.cs

```diff
@@ -0,0 +1,239 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Graph;
+
+namespace MicrosoftGraphSDK
+{
+    /// <summary>
+    /// Graph SDK endpoints for assignments
+    /// </summary>
+    public class Assignment
+    {
+        /// <summary>
+        /// Get the properties and relationships of an assignment
+        /// </summary>
+        /// <param name="client"></param>
+        /// <param name="classId"></param>
+        /// <param name="assignmentId"></param>
+        /// <returns>EducationAssignment</returns>
+        public static async Task<EducationAssignment> GetAssignment(
+            GraphServiceClient client,
+            string classId,
+            string assignmentId)
+        {
+            try
+            {
+                return await client.Education
+                    .Classes[classId]
+                    .Assignments[assignmentId]
+                    .Request()
+                    .GetAsync();
+            }
+            catch (Exception ex)
+            {
+                throw new GraphException($"GetAssignment call: {ex.Message}", classId, assignmentId);
+            }
+        }
+
+        /// <summary>
+        /// Get the properties and relationships of an assignment and pass a header value
+        /// </summary>
+        /// <param name="client"></param>
+        /// <param name="classId"></param>
+        /// <param name="assignmentId"></param>
+        /// <param name="headerName"></param>
+        /// <param name="headerValue"></param>
+        /// <returns>EducationAssignment</returns>
+        public static async Task<EducationAssignment> GetAssignment_WithHeader(
+            GraphServiceClient client,
+            string classId,
+            string assignmentId,
+            string headerName,
+            string headerValue)
+        {
+            try
+            {
+                return await client.Education
+                    .Classes[classId]
+                    .Assignments[assignmentId]
+                    .Request()
+                    .Header(headerName, headerValue)
+                    .GetAsync();
+            }
+            catch (Exception ex)
+            {
+                throw new GraphException($"GetAssignment_WithHeader call: {ex.Message}", classId, assignmentId, headerName, headerValue);
+            }
+        }
+
+        /// <summary>
+        /// Retrieve a list of assignment objects within a class
+        /// </summary>
+        /// <param name="client"></param>
+        /// <param name="classId"></param>
+        /// <returns>IEducationClassAssignmentsCollectionPage</returns>
+        public static async Task<IEducationClassAssignmentsCollectionPage> GetAssignments(
```

Comment:
all methods here must be "async" because they need to await for SDK response.

The "await" keyword is not needed in the workflow file, we just make the call to the method as usual